### PR TITLE
Add a BindingBase.debugBindingType() method to enable asserts that want to verify that the binding isn't initialized

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: packages/flutter/lib/src/foundation/binding.dart [hunk 1]
```diff
@@ -361,6 +364,32 @@ abstract class BindingBase {
...
+  /// before the point in the application code that wants to initialize the binding, or
+  /// to verify that the binding is the one that is expected.
+  ///
+  /// For example, if an application uses [Zone]s to report uncaught execptions, it may
+  /// need to ensure that `ensureInitialized()` has not yet been invoked on any binding
+  /// at the point where it configures the zone and initializes the binding.
+  ///
+  /// If this returns null, the binding has not been initialized.
+  ///
+  /// If this returns a non-null value, in returns the type of the binding instance.
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/binding.dart:379]: typo: in -> it

File: packages/flutter/lib/src/foundation/binding.dart [hunk 2]
```diff
@@ -361,6 +364,32 @@ abstract class BindingBase {
...
+  ///
+  /// To obtain the binding itself, consider the `instance` getter on the [BindingBase]
+  /// subclass or mixin.
+  ///
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/binding.dart:383]: Document that calling this is only valid in debug mode and will return null in non-debug mode even when the bindings are initialized?

File: packages/flutter/lib/src/foundation/binding.dart [hunk 3]
```diff
@@ -154,7 +154,7 @@ abstract class BindingBase {
   }
 
   bool _debugConstructed = false;
-  static bool _debugInitialized = false;
+  static Type? _debugInitializedType = null;
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/binding.dart:157]: lint is complaining about initializing this to null.

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/foundation/binding.dart (hunk 2 above)
Comment (Hixie) [reply packages/flutter/lib/src/foundation/binding.dart:383]: oh yikes, good call, i totally forgot to mention that

**Pull request merged**